codelists/scripts/update_nhs_refsets.py

```diff
@@ -147,26 +189,47 @@ def get_latest_db_release_and_refset_tag_from_api(base_url):
         resp = requests.get(url)
         resp.raise_for_status()
         data = resp.json()
-        if SNOMED_ID in data and "database_alias" in data[SNOMED_ID]:
-            database_alias = data[SNOMED_ID]["database_alias"]
+        if coding_system_id in data and "database_alias" in data[coding_system_id]:
+            database_alias = data[coding_system_id]["database_alias"]
             # Get latest refset tag if available, or default to a known version if not
-            latest_refset_tag = data.get("pcd_refsets", {}).get("tag", "20250627")
+            latest_refset_tag = data.get(tag_id, {}).get("tag", "20250627")
             return database_alias, latest_refset_tag
         else:
-            print(f"No database_alias found for {SNOMED_ID} in API response.")
-            return None
+            print(f"No database_alias found for {coding_system_id} in API response.")
+            return None, None
     except Exception as e:
```

Comment:
thought: I realise this code was here already, but maybe we can do better than a plain exception.

From looking at the code, it seems like the possible exceptions we might expect are from the external data access only:

* `requests.get()` (`requests.exceptions.ConnectionError`)
* `resp.raise_for_status()` (`requests.exceptions.HTTPError`)
* `resp.json()` (`requests.exceptions.JSONDecodeError`)

It might be possible to just use `requests.exceptions.RequestException` as a catch-all, but still narrow this down a bit.